File: cpp/src/benchmark/operation/RdmaWriteLatencyOperation.cpp
#include "RdmaWriteLatencyOperation.h"

namespace Observatory {

RdmaWriteLatencyOperation::RdmaWriteLatencyOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        RdmaLatencyOperation(benchmark, mode, operationCount, operationSize, Benchmark::RdmaMode::WRITE) {}

const char* RdmaWriteLatencyOperation::getClassName() const {
    return "RdmaWriteLatencyOperation";
}

const char* RdmaWriteLatencyOperation::getOutputFilename() const {
    return "RDMA Write Latency";
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/RdmaWriteLatencyOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Connection.Mode;
import de.hhu.bsinfo.observatory.benchmark.Connection.RdmaMode;

public class RdmaWriteLatencyOperation extends RdmaLatencyOperation {

    RdmaWriteLatencyOperation(Connection connection, Mode mode, int operationCount, int operationSize) {
        super(connection, mode, operationCount, operationSize, RdmaMode.WRITE);
    }

    @Override
    String getOutputFilename() {
        return "RDMA Write Latency";
    }
}
